Documentation/ApiOverview/Seo/PageTitleApi.rst

```diff
@@ -42,38 +44,67 @@ the page record will not be the correct title. To make sure to display the
-force you to have at least the :php:`getTitle()` method in your class. Within
+extending :php:`\TYPO3\CMS\Core\PageTitle\AbstractPageTitleProvider`.  Within
 this method you can create your own logic to define the correct title.
 
-..  code-block:: php
+..  literalinclude:: _ExampleSetInController/_MyOwnPageTitleProvider.php
+    :caption: EXT:my_extension/Classes/PageTitle/MyOwnPageTitleProvider.php
 
-    namespace Vendor\Extension\PageTitle;
+Usage example in an :ref:`Extbase <extbase>` controller:
 
-    use TYPO3\CMS\Core\PageTitle\AbstractPageTitleProvider;
+..  literalinclude:: _ExampleSetInController/_SomeController.php
+    :caption: EXT:my_extension/Classes/Controller/SomeController.php
 
-    class MyOwnPageTitleProvider extends AbstractPageTitleProvider
-    {
-        public function setTitle(string $title)
-        {
-            $this->title = $title;
-        }
-    }
+Configure the new page title provider in your TypoScript setup:
+
+..  literalinclude:: _ExampleSetInController/_setup.typoscript
+    :language: typoscript
+    :caption: EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript
+
+Example: Use values from the site configuration in the page title
+-----------------------------------------------------------------
+
+If you want to use data from the :ref:`site configuration <sitehandling>`, for
+example the site title, you can implement a page title provider as follows:
+
+..  literalinclude:: _ExampleWebsiteTitle/_WebsiteTitleProvider.php
+    :caption: EXT:my_sitepackage/Classes/PageTitle/WebsiteTitleProvider.php
+
+As we need to :ref:`inject <DependencyInjection>` the class :php:`SiteFinder`
+to retrieve the current site configuration, we must make the new page title
+provider public:
+
+..  literalinclude:: _ExampleWebsiteTitle/_Services.yaml
+    :language: yaml
+    :caption: EXT:my_sitepackage/Configuration/Services.yaml
+
+Then **flush the cache** in :guilabel:`Admin Tools > Maintenance > Flush TYPO3
+and PHP Cache`.
 
-Usage example, for example, in an :ref:`Extbase <extbase>` controller:
+Configure the new page title provider to be used in your TypoScript setup:
 
-.. code-block:: php
+..  literalinclude:: _ExampleWebsiteTitle/_setup.typoscript
+    :language: typoscript
+    :caption: EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript
 
-   $titleProvider = GeneralUtility::makeInstance(MyOwnPageTitleProvider::class);
-   $titleProvider->setTitle('Title from controller action');
+The registered page title providers are called after each other in the
+configured order. The first provider that returns a non-empty value is used,
+the providers later in the order are ignored.
 
+Therefore our custom provider should be loaded before `record`, the
+default provider which always returns a value. If the system extension
+:t3ext:`seo` is loaded the default :guilabel:`SEO Title` has a particular format,
+you can override this by loading your custom provider after `seo`.
```

Comment:
Hmm, but when I load it **after** seo and seo returns something, than my custom provider is ignored. So it should be **before** - or am I wrong?